src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/UpstageAIEmbeddingClient.java

```diff
@@ -59,7 +62,59 @@ Uni<EmbeddingResponse> embed(
 
     @ClientExceptionMapper
     static RuntimeException mapException(jakarta.ws.rs.core.Response response) {
-      return HttpResponseErrorMessageMapper.getDefaultException(response);
+      String errorMessage = getErrorMessage(response);
+      return HttpResponseErrorMessageMapper.mapToAPIException(
+          ProviderConstants.UPSTAGE_AI, response, errorMessage);
+    }
+
+    /**
+     * Extracts the error message from the response body. The example response body is:
+     *
+     * <pre>
+     * {
+     *   "message": "Unauthorized"
+     * }
+     *
+     * {
+     *   "error": {
+     *     "message": "This model's maximum context length is 4000 tokens. however you requested 10969 tokens. Please reduce your prompt.",
+     *     "type": "invalid_request_error",
+     *     "param": null,
+     *     "code": null
+     *   }
+     * }
+     * </pre>
+     *
+     * @param response The response body as a String.
+     * @return The error message extracted from the response body, or null if the message is not
+     *     found.
+     */
+    private static String getErrorMessage(jakarta.ws.rs.core.Response response) {
+      // Get the whole response body
+      String responseBody = response.readEntity(String.class);
+      JsonNode rootNode;
+      try {
+        rootNode = OBJECT_MAPPER.readTree(responseBody);
+      } catch (JsonProcessingException e) {
+        // should not go here, already check json format in EmbeddingProviderResponseValidation.
+        // if it happens, return the whole response body
+        return responseBody;
+      }
+      // Check if the root node contains a "message" field
+      JsonNode messageNode = rootNode.path("message");
+      if (!messageNode.isMissingNode()) {
+        return messageNode.asText();
+      }
+      // If the "message" field is not found, check for the nested "error" object
+      JsonNode errorNode = rootNode.path("error");
+      if (!errorNode.isMissingNode()) {
+        JsonNode errorMessageNode = errorNode.path("message");
+        if (!errorMessageNode.isMissingNode()) {
```

Comment:
In this case use JSON Pointer would simplify code a bit:

JsonNode errorMessageNode = rootNode.at("/error/message");
if (!errorMessageNode.isMissingNode()) {